dm/pkg/exstorage/utils.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exstorage
+
+import (
+	"context"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/br/pkg/storage"
+)
+
+// AdjustS3Path adjust s3 rawURL, add uniqueId into s3 path.
+func AdjustS3Path(rawURL string, uniqueID string) (string, error) {
+	if rawURL == "" {
+		return "", nil
+	}
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if u.Scheme == "s3" {
+		trimPath := strings.TrimRight(u.Path, "/")
+		// avoid duplicate add uniqueID
+		if uniqueID != "" && !strings.HasSuffix(trimPath, uniqueID) {
+			u.Path = trimPath + "." + uniqueID
+			return u.String(), nil
+		}
+		return rawURL, nil
+	}
+	return rawURL, nil
+}
+
+// isS3Path judges if rawURL is s3 path.
+func IsS3Path(rawURL string) (bool, error) {
+	if rawURL == "" {
+		return false, nil
+	}
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return false, err
+	}
+	if u.Scheme == "s3" {
```

Comment:
@glorv can we support use `https://s3.Region.amazonaws.com/bucket-name/key-name` to express a S3 URL?

Reply:
NO. the URL for s3 must be in format: `s3://bucketname/sub/object/path?...`.